### ISSUE TITLE
how do you simultaneously verify phone and email before you add the user to the user pool?

### ISSUE THREAD

**Issue by bionicles** [squidfunk/terraform-aws-cognito-auth#836]
## Description
for HIPAA stuff, how would we modify this to send verification email and verification text / SMS to phone simultaneously?

### Expected behavior

1. user enters phone + email in signup form
2. user receives phone + email verification concurrently
3. user must verify both to be added to the user pool

### Actual behavior

this project is highly relevant for us however it does not verify the phone before creating the user ... how can we do this BEFORE adding a user to our user pool?

note: verification after does not suit our use case because it would then be possible to deny another user access from their phone by using your email. 

TLDR: how do you implement simultaneous multifactor verification email+phone?

**Comment by squidfunk** [comment]
> TLDR: how do you implement simultaneous multifactor verification email+phone?

I haven't implemented something like this yet, so honestly, I don't know. However, note that the user is already added to the pool the moment when he completes the registration, but he can't sign in until he verifies his email address. This could possibly be solved with custom auth challenges or some other Cognito features, but it is beyond the scope of this project.

**Comment by bionicles** [comment]
:(

**Comment by squidfunk** [comment]
The only thing I can offer you in regard to your customization request is paid support, but I hope that you understand that the time that I can invest in open source for free is limited, as I have plenty of projects to maintain and only limited time beside my day job.

**Comment by squidfunk** [comment]
Housekeeping.
